store/store_test.go

```diff
@@ -268,6 +268,51 @@ func TestStore_Append_stableHeadWhenGaps(t *testing.T) {
 	}
 }
 
+func TestStoreGetByHeight_whenGaps(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	t.Cleanup(cancel)
+
+	suite := headertest.NewTestSuite(t)
+
+	ds := sync.MutexWrap(datastore.NewMapDatastore())
+	store := NewTestStore(t, ctx, ds, suite.Head(), WithWriteBatchSize(10))
+
+	head, err := store.Head(ctx)
+	require.NoError(t, err)
+	assert.Equal(t, head.Hash(), suite.Head().Hash())
+
+	firstChunk := suite.GenDummyHeaders(5)
+	_ = suite.GenDummyHeaders(5)
+	lastChunk := suite.GenDummyHeaders(5)
+
+	wantHead := firstChunk[len(firstChunk)-1]
+
+	{
+		err = store.Append(ctx, lastChunk...)
+		require.NoError(t, err)
+
+		shortCtx, shortCancel := context.WithTimeout(ctx, 10*time.Millisecond)
+		defer shortCancel()
+
+		head, err = store.GetByHeight(shortCtx, wantHead.Height())
+		require.Error(t, err)
```

Comment:
We need to check that GetByHeight succeeds as well

Height() == 100
goA GetByHeight(150) -> blocks
Append(200) -> no error
Append(range(100;200)) -> no error
goA -> no error